lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -128,7 +128,7 @@ def get_index(self, dataset: Dataset, root_index_name: AnyStr) -> AnyStr:
             index_map = Metadata.getvalue(dataset=dataset, key=Metadata.INDEX_MAP)
         except MetadataError:
             self.logger.error(
-                f"server.index-map not found in Metadata for a dataset {dataset!r}"
+                f"server.index-map not found in Metadata for a dataset {dataset!r} with key {Metadata.INDEX_MAP!r}"
```

Comment:
Now we're showing "server.index-map" *twice*; this should really be `f"Metadata key {Metadata.INDEX_MAP!r} not found for dataset {dataset!r}".

And really we shouldn't be pre-formatting f-strings with the `Logger`, so technically it should be

```
    self.logger.error("Metadata {} not found for dataset {}", Metadata.INDEX_MAP, dataset)
```

(The difference with `!r` here, in either form is that `str.__repr__` will add quotes around the key name, and the default `Dataset.__repr__` will be used instead of `Dataset.__str__` ... which I'm not actually sure I like, and I don't think the quotes are meaningful here.)